sw/device/lib/dif/dif_plic.c

```diff
@@ -168,6 +185,11 @@ static void plic_irq_pending_reg_info(dif_plic_irq_id_t irq,
   reg_info->bit_index = plic_reg_bit_index_from_irq_id(irq);
 }
 
+/**
+ * Get a total number of priority registers (one for every IRQ source).
+ */
+static size_t plic_num_priority_reg(void) { return kDifPlicIrqIdLast; }
```

Comment:
I think it is correct as it is, It is a bit confusing and nasty, I agree. The thing is that IRQ IDs start from `1` rather than `0`. And hence `79` is actual count of the IRQs, but when we loop through the registers we start from `0` as it makes the logic easier. I have plugged the actual values in the code snippet bellow:
```
  // Clear all of the priority registers.                                                                                                                                                      
  for (int i = 0; i < 79; ++i) {
    ptrdiff_t offset = RV_PLIC_PRIO0_REG_OFFSET + (i * sizeof(uint32_t));
    mmio_region_write32(plic->base_addr, offset, 0);
  }
```
Which makes the `offset` for the first IRQ register `RV_PLIC_PRIO0_REG_OFFSET + 0`, and the last one `RV_PLIC_PRIO0_REG_OFFSET + (78 * 4)`.

I think I will add a comment into the description of `plic_num_priority_reg`.

Reply:
Ok, so `PRIO0` is the priority register for `kDifPlicIrqIdGpio0`, which has ID of 1, i think?

Reply:
Yes, that is correct. It is a bit weird :/

Reply:
Remember: we are in full control of SW and HW. If there's a confusing naming going on, please file an issue to get that resolved for good; if you were confused by it, there's a high chance others will be equally confused.

Reply:
Well, it kind of makes sense, because you need some value for "nothing is pending" case. And `0` seems appropriate for that. I find the whole CC functionality slightly weird, but I think that is dictated by the PLIC spec?

Reply:
Without having followed the whole discussion: isn't the inconsistency that  the PRIO0 reg refers to interrupt 1? Couldn't this register be renamed? If that doesn't make sense, ignore; otherwise, file a follow-up.

Reply:
Hmmm, It does make sense. Thank you for pointing it out, I will have a think about it.

Reply:
Let's start a new issue to cover any changes to the `rv_plic` hardware, and in the meantime I think the logic here is good.

Reply:
#1823 is issue that I've brought up with the designers (along with two possible fixes).